### PR TITLE
Roe 1987 due diligence date error focus

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-1987

### Change description

Need to use individual field validation to get the the required click field focus. This is followed by setting further validation to check for multiple empty fields and proceed only if they have not been found. If they have been found we just pass over the validation and return true, as no further validation will be required.

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/validation/custom.validation.ts [hunk 1]
```diff
@@ -98,16 +98,18 @@ export const checkOptionalDate = (dayStr: string = "", monthStr: string = "", ye
 
 export const checkIdentityDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
   // to prevent more than 1 error reported on the date fields we check if the year is correct length or missing before doing the date check as a whole.
```
Comment (markpit) [inline src/validation/custom.validation.ts:100]: the comment will need updating as it says the length check is done first

Comment (mattch1) [reply src/validation/custom.validation.ts]: Updated comment

File: src/validation/custom.validation.ts [hunk 2]
```diff
@@ -81,78 +81,107 @@ const isYearEitherMissingOrCorrectLength = (yearStr: string = ""): boolean => {
...
-      const areAllDateFieldsPresent = checkAllDateFieldsArePresent(dayStr, monthStr, yearStr);
-      if (areAllDateFieldsPresent) {
-        const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
-        if (isDateValid) {
-          const isDatePastOrToday = checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
-          if (isDatePastOrToday) {
-            checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
-          }
-        }
+  // to prevent more than 1 error reported on the date fields we first check for multiple empty fields and then check if the year is correct length or missing before doing the date check as a whole.
```
Comment (mwestacott) [inline src/validation/custom.validation.ts:106]: I would update the comment on line 83 and 132 to match this comment change also. Keeps the comment consistent and it applies in all three cases.

**Review by mwestacott — APPROVED** [review]
Approving but please could comment be updated, as requested.

**Pull request merged**